ui/build.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+set -euv -o pipefail
+
+perform_push="${PERFORM_PUSH-false}"
+
+repository=shepmaster
+
+# Build outside the image so nodejs isn't installed on the main image
+cd frontend
+docker pull node
+docker run -it --rm -v $PWD:/ui --workdir /ui --entrypoint /bin/bash node -c '
+    yarn &&
+    NODE_ENV=production yarn run build
+'
+cd ..
+
+# Also don't put a rust compiler in the main playground image
+docker run -it --rm -v $PWD:/ui --workdir /ui --entrypoint /bin/bash shepmaster/rust-nightly -c '
+    rustup target add x86_64-unknown-linux-musl &&
+    cargo build --target=x86_64-unknown-linux-musl --release
```

Comment:
With my proposed change to reuse the S3-uploaded artifacts, this file might not make sense as `ui/build.sh` anymore. It could just be moved to the `.travis/` directory.